modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/circuit/SeqexecCircuit.scala

```diff
@@ -127,43 +127,31 @@ object SeqexecCircuit extends Circuit[SeqexecAppRootModel] with ReactConnector[S
     zoom(_.uiModel.configTableState)
 
   val sequencesOnDisplayRW: ModelRW[SeqexecAppRootModel, SequencesOnDisplay] =
-    this.zoomRWL(SeqexecAppRootModel.uiModel ^|-> SeqexecUIModel.sequencesOnDisplay)
+    this.zoomRWL(SeqexecAppRootModel.sequencesOnDisplayL)
 
-  def sequenceTab(id: Observation.Id): ModelR[SeqexecAppRootModel, SeqexecTabActive] =
-    this.zoomG((SeqexecAppRootModel.uiModel ^|-> SeqexecUIModel.sequencesOnDisplay).composeGetter(SequencesOnDisplay.tabG(id)))
+  def sequenceTab(id: Observation.Id): ModelR[SeqexecAppRootModel, Option[SeqexecTabActive]] =
+    this.zoomG(SeqexecAppRootModel.sequencesOnDisplayL.composeGetter(SequencesOnDisplay.tabG(id)))
 
-  def sequenceObserverReader(id: Observation.Id): ModelR[SeqexecAppRootModel, SequenceInfoFocus] =
-    statusReader.zip(sequenceTab(id)).zoom {
-      case (status, SeqexecTabActive(tab, _)) =>
-        val targetName = tab.sequence.flatMap(firstScienceStepTargetNameT.headOption)
-        SequenceInfoFocus(status.isLogged, tab.sequence.map(_.metadata.name), tab.sequence.map(_.status), targetName)
-    }
+  def sequenceObserverReader(id: Observation.Id): ModelR[SeqexecAppRootModel, Option[SequenceInfoFocus]] =
+    this.zoomG(SequenceInfoFocus.sequenceInfoG(id))
 
   def statusAndStepReader(id: Observation.Id): ModelR[SeqexecAppRootModel, Option[StatusAndStepFocus]] =
-    statusReader.zip(sequenceTab(id)).zoom {
-      case (status, SeqexecTabActive(tab, _)) =>
-        tab.sequence.map { t =>
-          StatusAndStepFocus(status.isLogged, t.metadata.instrument, t.id, tab.stepConfigDisplayed, t.steps.length, tab.isPreview)
-        }
-    }
+    this.zoomG(StatusAndStepFocus.statusAndStepG(id))
```

Comment:
i'm aiming to remove diode's `ModalR` which is kind of a lens with proper Monocle lenses